use-cases.md

```diff
@@ -53,11 +53,7 @@ When a server wants to trace execution of a request, it generally needs to go th
 Let's assume that we have an HTTP server, and the Span is propagated from the client via HTTP headers, accessible via `request.headers`:
 
 ```python
-    span = tracer.join_trace_from_text(
-        operation_name=operation,
-        context_snapshot=request.headers,
-        trace_attributes=request.headers
-    )
+    span = tracer.get_span_extractor(request.class).extract(request)
```

Comment:
imo this makes the OT implementation work much harder than it needs to. For the end user it looks like magic, and implementation needs to know about 100s of various rpc request types. I don't think it's a useful way to spend implementors time chasing after all different ways RPC frameworks express their requests and learning those APIs, not to mention that the classpath dependency of a tracer will become huge.

In the current API, the OT implementor only needs to support two representations, a binary blob and a string key/value map. End users (RPC implementors) can use those two representations and inject them into their own Request class using their own API, without making it the responsibility of the OT tracer.